internal/test/e2e/kubectl.go

```diff
@@ -165,3 +177,53 @@ func NamespaceForTest(t *testing.T, kubectl *kubectlContext) string {
 
 	return name
 }
+
+// EventuallyInCluster runs a command as a Job in the current Kubernetes cluster namespace.
+func EventuallyInCluster(kubectl *kubectlContext, name string, deadline time.Duration, image string, command ...string) (string, error) {
+	job := &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Job",
+			APIVersion: "batch/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:    "container1",
+							Image:   image,
+							Command: command,
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyOnFailure,
+				},
+			},
+		},
+	}
+
+	f, err := ioutil.TempFile("", "e2e-job."+name+".json")
+	if err != nil {
+		return "", err
+	}
+	defer os.Remove(f.Name())
+	defer f.Close()
+
+	encoder := json.NewEncoder(f)
+	err = encoder.Encode(job)
+	if err != nil {
+		return "", err
+	}
+	err = kubectl.Apply("--filename", f.Name())
```

Comment:
i think you can pipe input into `kubectl apply`, might be nicer than creating a file on disk?

Reply:
I agree, that would be neater, but I'm going to leave it for now. It would require some changes to the `kubectl` library to allow us to set stdin.